torch/csrc/jit/passes/quantization/quantization_patterns.h

```diff
@@ -616,6 +616,30 @@ graph(%a_quant, %min, %max):
           %r = aten::clamp(%a_quant, %min, %max)
           return (%r) )";
 
+  // aten::hardtanh
+  std::string hardtanh = R"(
+graph(%a_quant, %min, %max):
+          %a_dequant = aten::dequantize(%a_quant)
+          %r = aten::hardtanh(%a_dequant, %min, %max)
```

Comment:
For hardtanh, the scale and zero-point are determined based on the min/max arguments to aten. There are a subset of ops where the scale and zero-point are functions of the op itself (not dependent on the input tensor) (hardtanh for example).  We also have a larger set (avg_pool, interpolate where we are assuming that the output scale is the same as that of the input). Do we distinguish between these two types?

Reply:
qparams for hardtanh/clamp are inherited from input tensor: https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/quantized/cpu/qclamp.cpp#L48.

Yes, I have another PR coming up to handle the ops that have fixed qparams.